src/asInput/index.jsx

```diff
@@ -46,6 +46,14 @@ const asInput = (WrappedComponent, labelFirst = true) => {
       };
     }
 
+    componentWillReceiveProps(nextProps) {
+      if (nextProps.value !== this.props.value) {
```

Comment:
do we want to compare `nextProps` against `this.props.value` or `this.state.value`?

Reply:
I think we want to compare it with props, per https://reactjs.org/docs/react-component.html#componentwillreceiveprops:
```
If you need to update the state in response to prop changes (for example, to reset it), you may compare this.props and nextProps and perform state transitions using this.setState() in this method.
```

Reply:
So imagine that I have a `render` like this

```javascript
render() {
  return <InputText value={this.state.emailAddress}>;
}
```
As I type into this component, `state.value` of `myInput` will change via the `handleChange` method. 

But if `this.state.emailAddress` does not change (for example, there's no `onChange` handler) then `nextProps.value` will equal `this.props.value` which means that the `state.value` of `NewComponent` will not reflect the value of the `value` prop.

Reply:
@jaebradley I believe this should be props. Do you have a story that tests that hypothetical and shows that's it doesn't work? I think it'll be easier to discuss this more if it's not a conceptual example and something tangible to poke at.

Reply:
yeah this looks good after all - I was confusing what `value` means; now it kind've means initial value or "reset" value.